step-functions/step-functions-client-api/src/main/java/step/functions/type/AbstractFunctionType.java

```diff
@@ -275,31 +275,24 @@ protected FileVersionCloseable registerFile(DynamicValue<String> dynamicValue, S
 	protected FileVersionCloseable registerFile(File file, String propertyName, Map<String, String> props, boolean cleanable) {
 		FileVersion fileVersion = registerFile(file, cleanable);
 		registerFileVersionId(propertyName, props, fileVersion.getVersionId());
-		return new FileVersionCloseable(fileVersion);
+		return new FileVersionCloseable(fileVersion, gridFileServices);
 	}
 
-	/**
-	 * Release the provided file version in the grid's file manager. Should be called by the caller registering the file version once it doesn't require it anymore
-	 *
-	 * @param fileVersion the {@link FileVersion} to be released
-	 */
-	private void releaseFile(FileVersion fileVersion) {
-		if (fileVersion != null) {
-			gridFileServices.releaseFile(fileVersion);
-		}
-	}
-
-	protected class FileVersionCloseable implements AutoCloseable {
+	protected static class FileVersionCloseable implements AutoCloseable {
 
-		public FileVersion fileVersion;
+		protected final FileVersion fileVersion;
+		protected final GridFileService gridFileServices;
 
-		private FileVersionCloseable(FileVersion fileVersion) {
+		private FileVersionCloseable(FileVersion fileVersion, GridFileService gridFileServices) {

```

Comment:
You can Objects.requireNonNull on both arguments, as (at least currently) all code paths using the constructor guarantee that both arguments are non-null. Anyway, it doesn't make sense to accept a null fileVersion, as there is nothing to do in that case, so the "registration" should be skipped altogether.